model/task/task.go

```diff
@@ -1105,36 +1105,72 @@ func (t *Task) markAsHostDispatchedWithFunc(doUpdate func(update bson.M) error,
 	t.AgentVersion = agentRevision
 	t.LastHeartbeat = dispatchTime
 	t.DistroId = distroID
+	t.Aborted = false
+	t.AbortInfo = AbortInfo{}
+	t.Details = apimodels.TaskEndDetail{}
 
 	return nil
 }
 
-// MarkAsHostUndispatched marks that the host task is undispatched. If the task
-// is already dispatched to a host, it unsets the host ID field on the task. It
-// returns an error if any of the database updates fail.
+// MarkAsHostUndispatchedWithContext marks that the host task is undispatched.
+// If the task is already dispatched to a host, it aborts the dispatch by
+// undoing the dispatch updates. This is the inverse operation of
+// MarkAsHostDispatched.
 func (t *Task) MarkAsHostUndispatched() error {
-	// then, update the task document
-	t.Status = evergreen.TaskUndispatched
+	doUpdate := func(update bson.M) error {
+		return UpdateOne(bson.M{IdKey: t.Id}, update)
+	}
+	if err := t.markAsHostUndispatchedWithFunc(doUpdate); err != nil {
+		return err
+	}
 
-	return UpdateOne(
-		bson.M{
-			IdKey: t.Id,
+	event.LogHostTaskUndispatched(t.Id, t.Execution, t.HostId)
+
+	return nil
+}
+
+// MarkAsHostUndispatchedWithContext marks that the host task is undispatched.
+// If the task is already dispatched to a host, it aborts the dispatch by
+// undoing the dispatch updates. This is the inverse operation of
+// MarkAsHostDispatchedWithContext.
```

Comment:
I think it's okay to leave it as is since other transaction-related updates like `MarkAsHostDispatchedWithContext` don't mention the event log. I included it for the host updates only because they have a non-transaction equivalent that does something different.

Reply:
Ah I see. Ok